magma/circuit.py

```diff
@@ -235,6 +235,7 @@ def __new__(metacls, name, bases, dct):
         dct.setdefault('primitive', False)
         dct.setdefault('coreir_lib', 'global')
         dct.setdefault("inline_verilog_strs", [])
+        dct.setdefault("coreir_metadata", {})
```

Comment:
it's a bit cheeky, but we could consider adding a set of predefined dicts in `__prepare__` so that the code could look like:

```python
class Foo(m.Circuit):
    coreir_metadata["inline_single_instances"] = False
    ...
```

it just kinda looks a little nicer but not sure it's worth it :)

Reply:
Clever, I wonder if this has any problematic interactions with subclassing?
I think in subclassing you would need to do something like
```
class Bar(Foo):
    coreir_metadata = Foo.coreir_metadata
```
to inherit class variables so doesn't seem like it would be an issue since you need to explicitly do it in Python (and this would override the prepare introduced dictionary)

Reply:
I guess from that perspective it introduces an implicit class variable which is not a typical pattern in Python so the downside is losing Pythonic style.

Reply:
Merging this as a first pass since if we decide to add the __prepare__ logic, it should be backwards compatible (this syntax will just overwrite the prepare dict which is totally fine).